src/peppr/evaluator.py

```diff
@@ -79,6 +80,58 @@ def metrics(self) -> tuple[Metric, ...]:
     def system_ids(self) -> tuple[str, ...]:
         return tuple(self._ids)
 
+    @staticmethod
+    def combine(evaluators: Iterable["Evaluator"]) -> "Evaluator":
+        """
+        Combine multiple :class:`Evaluator` instances into a single one,
+        preserving the systems fed to each instance.
+
+        Parameters
+        ----------
+        evaluators : Iterable of Evaluator
+            The evaluators to combine.
+            The ``metrics``, ``tolerate_exceptions`` and ``min_sequence_identity``
+            must be the same for all evaluators.
+
+        Returns
+        -------
+        Evaluator
+            The evaluator combining the systems of all input `evaluators` in the order
+            of the input.
+        """
+        is_same = True
+        ref_evaluator = None
+        all_ids = []
+        all_results = []
+        for evaluator in evaluators:
+            if ref_evaluator is None:
+                ref_evaluator = evaluator
+            if len(evaluator._metrics) != len(ref_evaluator._metrics):
+                is_same = False
+            for metric, ref_metric in zip(evaluator._metrics, ref_evaluator._metrics):
+                if metric.name != ref_metric.name:
+                    is_same = False
+            if evaluator._tolerate_exceptions != ref_evaluator._tolerate_exceptions:
+                is_same = False
+            if evaluator._min_sequence_identity != ref_evaluator._min_sequence_identity:
+                is_same = False
+            if not is_same:
```

Comment:
Maybe `__eq__` doesn't make sense here because we're not additionally verifying the equality of the evaluator's systems themselves, but it could result in slightly more idiomatic python. I think there's a reasonable argument for equality only being relevant to the parameterization of the container object (the evaluator in this case) itself, while not extending to its contents (the systems).